schema/parser.go

```diff
@@ -413,6 +413,19 @@ func parseIdentity(opt *sqlparser.IdentityOpt) string {
 	return strings.ToUpper(opt.Behavior)
 }
 
+func parseDefaultDefinition(opt *sqlparser.DefaultDefinition) *DefaultDefinition {
+	if opt == nil || opt.Value == nil {
+		return nil
+	}
+	defaultVal := parseValue(opt.Value)
+
+	name := "DEFAULT"
+	if opt.Name.String() != "" {
+		name = opt.Name.String()
```

Comment:
question: which test of mssqldef_test cover this path? I couldn't figure out why this is needed.

Reply:
The test named `TestMsqldeffCreateTableDropColumnWithDefault` covers this case.
This is necessary to resolve the constraint name of the default constraint.

A default constraint name like `DF__users__name__XXXXXX` will be generated automatically when the default value is set in the sql server as follows
```
CREATE TABLE users (
  id bigint NOT NULL PRIMARY KEY,
  name varchar(20) DEFAULT NULL
);
```
And in order to drop the default constraint, I need to run the following query.
```
ALTER TABLE users DROP CONSTRAINT DF__users__name__XXXXXX
```
So I had to parse the default constraint name here.

Reply:
Thanks for the clarification. 

So `DefaultDefinition#name` is actually a name of a constraint. If this part is about a constraint, I feel the design is again a bit inconsistent with the existing code. For example, when you have `UNIQUE` in a column definition, we do NOT put that information as a part of the column, but add a unique index object to the table. Do you think a similar design (add a constraint object to the table) makes more sense to the constraints as well? Is there any way in mssql to specify a default constraint?

<hr>

Another question: Is it possible in mssql to explicitly specify a constraint name? If so, can we add a test case for it? If such a usage doesn't really makes sense (you never want to rename `DF__users__name__XXXXX` to something easier to manage/understand), never mind.

<hr>

One more question: As you can see in psqldef_test, we already have support for `CONSTRAINT`. And it's implemented as a `ForeignKey` object. Do you think it makes sense for mssql to consider constraints as foreign keys, or are constraints completely a different thing from foreign keys in mssql?

Reply:
> Do you think a similar design (add a constraint object to the table) makes more sense to the constraints as well? Is there any way in mssql to specify a default constraint?

In mssql, I cannot define a default constraint as `<table_constraint>`. I need to define default constraint as `<column_definition>`. This is what makes it different from the other `PRIMARY KEY`, `UNIQUE`, `FOREIGN KEY`, and `CHECK` constraints. So I needed a specific implementation only for the default constraints, which is what I did in this pull request.
The following document is clear about the definition of mssql constraints.
https://docs.microsoft.com/en-us/sql/t-sql/statements/create-table-transact-sql?view=sql-server-2017#syntax-for-memory-optimized-table
<hr>

> Is it possible in mssql to explicitly specify a constraint name? 

I can write the following
```
CREATE TABLE users (
  id bigint NOT NULL PRIMARY KEY,
  name varchar(20) CONSTRAINT df_name DEFAULT NULL
);
```
I agree that this way makes the test code clearer.
<hr>

> Do you think it makes sense for mssql to consider constraints as foreign keys, or are constraints completely a different thing from foreign keys in mssql?

Also in mssql, foreign key can be considered as one of the constraints.
I plan to define `PRIMARY KEY` and `UNIQUE`, `FOREIGN KEY` and `CHECK` as `<table_constraint>` as well as the other DBs.
When dropping a column, it is necessary to find the index of the corresponding column from the index information and remove the constraint in the same way that I remove the constraint of the primary key in this pull request.
https://github.com/k0kubun/sqldef/pull/123/files#diff-d30ef67aecc70d57c316863ba09cbb56a864bc450da7db94a539dadcd81f4ee0R217

Reply:
> In mssql, I cannot define a default constraint as <table_constraint>. I need to define default constraint as <column_definition>

Got it. Then let's have a default constraint as a part of a column definition as you did, unlike other constraints.

> I can write the following
> I agree that this way makes the test code clearer.

Unless you intend to always use the explicit version, can we actually test both cases? If possible, I want `replaceAutoNamedConstraint` to be kept deleted to avoid misleading readers.

> I plan to define PRIMARY KEY and UNIQUE, FOREIGN KEY and CHECK as <table_constraint> as well as the other DBs.

:+1: